hibernate-core/src/main/java/org/hibernate/hql/internal/ast/tree/FromElement.java

```diff
@@ -348,9 +349,7 @@ public String getIdentityColumn() {
 					table, propertyNames[i],
 					inSelect
 			);
-			for ( int j = 0; j < propertyNameColumns.length; j++ ) {
-				columns.add( propertyNameColumns[j] );
-			}
+			columns.addAll(Arrays.asList(propertyNameColumns));
```

Comment:
There is a better way so that a new Arrays.asList wrapper won't be allocated. java.util.Collections.addAll();

Reply:
That's right. Feel free to provide an update.

Reply:
Because this change affects code in the HQL parser and entity persisters, I think the extra memory allocation is unacceptable. I am sure there's a reason why these were loops over arrays in the first place 😄  Changing the code to use `java.util.Collections.addAll` however may improve readability.

Reply:
Problem with a loop is that it can cause repeating acquisition of memory while processing the loop which increases pressure on the garbage collection. In general it is better acquire the needed memory only once.

Reply:
@carstenartur what "repeating acquisition of memory" are you referring to? I guess the Collection needing to be resized?

It sould be possible to hint for the resize just once before looping, and yet avoid allocating an intermediary `List` instance.

Reply:
@Sanne Yes, I talk about resizing.

Reply:
for example:

     ArrayList<String> columns = new ArrayList<>();
    ...
     columns.ensureCapacity( columns.size() + propertyNameColumns.length );
     Collections.addAll( columns, propertyNameColumns );

Or even better, would be lovely to get rid of the ArrayList and the intermediate `propertyNameColumns` arrays, but that's certainly going to get a bit more complex and is unlikely to be worth is as this is all being rewritte in in 6.
I would suggest the above change so to stick to the intent of this commit: getting rid of manual array copies over a loop, which is a very sensible idea.

Thanks!

Reply:
Using ensureCapacity is the way to go here IMO